src/Geometry/localgeometry.jl

```diff
@@ -18,28 +18,52 @@ struct LocalGeometry{I, C <: AbstractPoint, FT, S}
     "inverse Jacobian"
     invJ::FT
     "Partial derivatives of the map from `ξ` to `x`: `∂x∂ξ[i,j]` is ∂xⁱ/∂ξʲ"
-    ∂x∂ξ::Axis2Tensor{FT, Tuple{LocalAxis{I}, CovariantAxis{I}}, S}
+    ∂x∂ξ::∂x∂ξT #::Axis2Tensor{FT, Tuple{LocalAxis{I}, CovariantAxis{I}}, S}
     "Partial derivatives of the map from `x` to `ξ`: `∂ξ∂x[i,j]` is ∂ξⁱ/∂xʲ"
-    ∂ξ∂x::Axis2Tensor{FT, Tuple{ContravariantAxis{I}, LocalAxis{I}}, S}
+    ∂ξ∂x::∂ξ∂xT #::Axis2Tensor{FT, Tuple{ContravariantAxis{I}, LocalAxis{I}}, S}
     "Contravariant metric tensor (inverse of gᵢⱼ), transforms covariant to contravariant vector components"
-    gⁱʲ::Axis2Tensor{FT, Tuple{ContravariantAxis{I}, ContravariantAxis{I}}, S}
+    gⁱʲ::gⁱʲT #::Axis2Tensor{FT, Tuple{ContravariantAxis{I}, ContravariantAxis{I}}, S}
     "Covariant metric tensor (gᵢⱼ), transforms contravariant to covariant vector components"
-    gᵢⱼ::Axis2Tensor{FT, Tuple{CovariantAxis{I}, CovariantAxis{I}}, S}
-    @inline function LocalGeometry(
+    gᵢⱼ::gᵢⱼT #::Axis2Tensor{FT, Tuple{CovariantAxis{I}, CovariantAxis{I}}, S}
+end
+
+full_lg_type(I, C, FT, S) = LocalGeometry{
+    I,
+    C,
+    FT,
+    Axis2Tensor{FT, Tuple{LocalAxis{I}, CovariantAxis{I}}, S},
+    Axis2Tensor{FT, Tuple{ContravariantAxis{I}, LocalAxis{I}}, S},
+    Axis2Tensor{FT, Tuple{ContravariantAxis{I}, ContravariantAxis{I}}, S},
+    Axis2Tensor{FT, Tuple{CovariantAxis{I}, CovariantAxis{I}}, S},
+}
+
+@inline function LocalGeometry(
+    coordinates,
+    J,
+    WJ,
+    ∂x∂ξ::Axis2Tensor{FT, Tuple{LocalAxis{I}, CovariantAxis{I}}, S},
+) where {FT, I, S}
+    ∂ξ∂x = inv(∂x∂ξ)
+    C = typeof(coordinates)
+    Jinv = inv(J)
+    gⁱʲ = ∂ξ∂x * ∂ξ∂x'
+    gᵢⱼ = ∂x∂ξ' * ∂x∂ξ
+    isapproxsymmetric(components(gⁱʲ)) || error("gⁱʲ is not symmetric.")
+    isapproxsymmetric(components(gᵢⱼ)) || error("gᵢⱼ is not symmetric.")
+    ∂x∂ξT = typeof(∂x∂ξ)
+    ∂ξ∂xT = typeof(∂ξ∂x)
+    gⁱʲT = typeof(gⁱʲ)
+    gᵢⱼT = typeof(gᵢⱼ)
+    return full_lg_type(I, C, FT, S)(
```

Comment:
```suggestion
    return LocalGeometry(
```
All of the type parameters should be inferable now.